Assets/Scripts/MVC/Bullets/BulletView.cs

```diff
@@ -9,41 +9,29 @@ namespace Outscal.BattleTank
     public class BulletView : MonoBehaviour
     {
         public BulletController bulletController { get; private set; }
-        public ParticleSystem BullectDestroyVFX;
         public float m_MaxLifeTime = 1f;
 
-        //void Start()
-        //{
-        //    // If it isn't destroyed by then, destroy the shell after it's lifetime.
-        //    Destroy(gameObject, m_MaxLifeTime);
-        //}
-
         private void FixedUpdate()
         {
             bulletController.Movement();
         }
 
+        public void SetBulletController(BulletController _bulletController)
+        {
+            bulletController = _bulletController;
+        }
+
         private void OnCollisionEnter(Collision collision)
         {
             if (collision.gameObject.GetComponent<EnemyTankView>() != null)
             {
-                Destroy(collision.gameObject);
-                this.gameObject.SetActive(false);
+                collision.gameObject.GetComponent<EnemyTankView>().enemyTankController.ApplyDamage(bulletController.bulletModel.Damage);
```

Comment:
EnemyTankView enemyTankView =  collision.gameObject.GetComponent<EnemyTankView>();

use like this

Reply:
you can use compare tag .
so much dots is not a good approach